library/ssl_tls13_generic.c

```diff
@@ -966,6 +982,143 @@ int mbedtls_ssl_tls13_write_certificate( mbedtls_ssl_context *ssl )
 /*
  * STATE HANDLING: Output Certificate Verify
  */
+static int ssl_tls13_get_sig_alg_from_pk( mbedtls_ssl_context *ssl,
+                                          mbedtls_pk_context *own_key,
+                                          uint16_t *algorithm,
+                                          mbedtls_pk_type_t *pk_type,
+                                          mbedtls_md_type_t *md_alg)
+{
+    mbedtls_pk_type_t sig = mbedtls_ssl_sig_from_pk( own_key );
+    /* Determine the size of the key */
+    size_t own_key_size = mbedtls_pk_get_bitlen( own_key );
+    *algorithm = MBEDTLS_TLS1_3_SIG_NONE;
+    ((void) own_key_size);
+
+    switch( sig )
+    {
+#if defined(MBEDTLS_ECDSA_C)
+        case MBEDTLS_SSL_SIG_ECDSA:
+            switch( own_key_size )
+            {
+                case 256:
+                    *algorithm = MBEDTLS_TLS1_3_SIG_ECDSA_SECP256R1_SHA256;
+                    *md_alg = MBEDTLS_MD_SHA256;
+                    *pk_type = MBEDTLS_PK_ECDSA;
+                    return( 0 );
+                case 384:
+                    *algorithm = MBEDTLS_TLS1_3_SIG_ECDSA_SECP384R1_SHA384;
+                    *md_alg = MBEDTLS_MD_SHA384;
+                    *pk_type = MBEDTLS_PK_ECDSA;
+                    return( 0 );
+                case 521:
+                    *algorithm = MBEDTLS_TLS1_3_SIG_ECDSA_SECP521R1_SHA512;
+                    *md_alg = MBEDTLS_MD_SHA512;
+                    *pk_type = MBEDTLS_PK_ECDSA;
+                    return( 0 );
+                default:
+                    MBEDTLS_SSL_DEBUG_MSG( 3,
+                                           ( "unknown key size: %"
+                                             MBEDTLS_PRINTF_SIZET " bits",
+                                             own_key_size ) );
+                    break;
+            }
+            break;
+#endif /* MBEDTLS_ECDSA_C */
+
+#if defined(MBEDTLS_RSA_C)
```

Comment:
This code is related to Certificate Verify for which we may need to compute an RSA PSS signature without x509 Mbed TLS code being involved thus the couple (MBEDTLS_RSA_C, MBEDTLS_PKCS1_V21) is indeed here better than MBEDTLS_X509_RSASSA_PSS_SUPPORT to me.

Reply:
It has been guards with RSA_C and PKCS1_V21